hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java

```diff
@@ -197,13 +200,35 @@ public boolean checkAndEnforceThrottle(
         if (manager == null) {
             return true;
         }
-        if (!manager.allReqsMetAt(now)) {
+
+        final boolean allReqMet;
+        if (queryFunction == CRYPTO_GET_ACCOUNT_BALANCE
+                && configuration.getConfigData(TokensConfig.class).countingGetBalanceThrottleEnabled()) {
+            final var accountStore = new ReadableStoreFactory(state).getStore(ReadableAccountStore.class);
+            final int balanceCount = getBalanceCount(query, accountStore);
+            allReqMet = manager.allReqsMetAt(now, Math.max(1, balanceCount), ONE_TO_ONE);
+        } else {
+            allReqMet = manager.allReqsMetAt(now);
+        }
+
+        if (!allReqMet) {
             reclaimLastAllowedUse();
             return true;
         }
         return false;
     }
 
+    private int getBalanceCount(@NonNull final Query query, @NonNull final ReadableAccountStore accountStore) {
+        final var accountID = query.cryptogetAccountBalanceOrThrow().accountID();
+        if (accountID != null) {
+            final var account = accountStore.getAccountById(accountID);
+            if (account != null) {
+                return account.numberPositiveBalances();
```

Comment:
An account with lots of associated `0` token balances will still be expensive to query; I would suggest,
```
return Math.min(account.numberAssociations(), tokenConfig.maxRelsPerInfoQuery());
```
(C.f. [here](https://github.com/hashgraph/hedera-services/blob/develop/hapi/hedera-protobufs/services/state/token/account.proto#L132) and [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountBalanceHandler.java#L186)).

Reply:
Right, good catch! I will fix it.